labonneboite/web/search/views.py

```diff
@@ -282,7 +284,7 @@ def entreprises():
     )
 
     # Stop here in case of invalid arguments
-    if not form.validate() or job_doesnt_exist:
+    if location_doesnt_exist or job_doesnt_exist or not form.validate():
```

Comment:
Il y a un truc que je ne comprends pas : comment ça se fait que la validation du formulaire plante quand on passe des arguments du mauvais type ? Pour moi c'est le formulaire qui devrait ne pas se valider, et puis basta.

Reply:
j'ai une piste sur https://stackoverflow.com/questions/15366452/wtforms-how-to-use-a-decimalfield-which-generates-proper-field-errors-on-bad-inp

How are you providing the form data to the form (how are you constructing it)? is it being provided using a form-data multi-dict? If so, this error shouldn't be occurring, as the data should never be coerced over to the .data property. This usually happens when people pass raw form data as **kwargs, which are supposed to be only for already-coerced data. – Crast Apr 11 '13 at 21:41

on utilise effectivement `CompanySearchForm(**kwargs)`

je vais tenter d'utiliser un `MultiDict`

WIP ...